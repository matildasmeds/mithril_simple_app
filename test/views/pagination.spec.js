var m = require("mithril");
var tidy = require("mithril-jest").tidy;

var getDataFn = function(opts) {
    return Promise.resolve(opts);
};

var Pagination = require("../../src/views/Pagination").for(getDataFn);

var testResultFn = function(result) {
    var view = m(result);
    var html = tidy(view);
    expect(html).toMatchSnapshot();
}

describe("Pagination", function() {
    it("Creates pagination links", function() {
	Pagination.getData({
	    total: 51,
	    limit: 15,
	    offset: 30,
	}).then(testResultFn);
    });
    it("Shows no links, when just one page", function() {
	Pagination.getData({
	    total: 100,
	    limit: 101,
	    offset: 0,
	}).then(testResultFn);
    });
});


