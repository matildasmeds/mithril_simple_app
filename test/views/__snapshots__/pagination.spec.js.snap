exports[`Pagination Shows no links, when just one page 1`] = `
"<div class=\"pagination-region\">
  <div id=\"pagination-links\" class=\"pagination-links\">
    <a id=\"link-to-page-1\" class=\"pagination-link\">1</a><a id=
    \"link-to-page-2\" class=\"pagination-link\">2</a><a id=
    \"link-to-page-3\" class=\"pagination-link active\">3</a><a id=
    \"link-to-page-4\" class=\"pagination-link\">4</a>
  </div>
</div>
"
`;

exports[`Pagination Shows no links, when just one page 2`] = `
"<div class=\"pagination-region\">
  <div id=\"pagination-links\" class=\"pagination-links\"></div>
</div>
"
`;
